src/components/comment-section/CommentSection.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from 'react';
+import { Comment } from '../../types/types';
+import './CommentSection.css';
+
+type CommentSectionProps = {
+  currentUserId: number;
+  currentUserName: string;
+  initialComments: Comment[] | undefined;
+  onCommentsChange: (comments: Comment[]) => void;
+};
+
+const CommentSection: React.FC<CommentSectionProps> = ({
+  currentUserId,
+  currentUserName,
+  initialComments,
+  onCommentsChange,
+}) => {
+  const [comments, setComments] = useState<Comment[] | undefined>(
+    initialComments,
+  );
+  const [showCommentInput, setShowCommentInput] = useState(false);
+  const [newComment, setNewComment] = useState('');
+
+  useEffect(() => {
+    setComments(initialComments);
+  }, [initialComments]);
+
+  const handleNewCommentClick = () => {
+    setShowCommentInput(true);
+  };
+
+  const handleCommentChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setNewComment(e.target.value);
+  };
+
+  const handleSendComment = () => {
+    if (newComment.trim()) {
+      const comment: Comment = {
+        id: Date.now(),
```

Comment:
please do not use this for id, what you can do is to include additional library 'uuid' and used it as identifier